testing/test_tmpdir.py

```diff
@@ -400,11 +401,13 @@ def test_on_rm_rf_error(self, tmp_path: Path) -> None:
             assert fn.is_file()
 
         # ignored function
-        with pytest.warns(None) as warninfo:
-            exc_info4 = (None, PermissionError(), None)
-            on_rm_rf_error(os.open, str(fn), exc_info4, start_path=tmp_path)
-            assert fn.is_file()
-        assert not [x.message for x in warninfo]
+        with warnings.catch_warnings():
```

Comment:
should this test use a mix of warns/catch_warnings to begin with?

imho there should be s single context manager which ensures no warning happens
but if  it shows tricky to spell different quickly its fine to make that a follow-up task

Reply:
That would be nice, but I am not sure if it is possible or how to do it. 
As I understand we want to discourage folks from using `None`, but we are maintaining the functionality (and this test) only for backwards-compatibility. I would also be inclined to say that this test case could be removed(!), so maybe not super important to worry about it?

Reply:
> should this test use a mix of warns/catch_warnings to begin with?

It's a weird test, but I think that Olga's edits are good - we'll remove/change it in 7.0 when the `warns(None)` deprecation is turned into an error.

> imho there should be s single context manager which ensures no warning happens
but if it shows tricky to spell different quickly its fine to make that a follow-up task

IMO this is probably served with `with warnings.catch_warnings():` (in the rare times that `-Werror` isn't enough); and if not it's a separate issue rather than aditional work on this PR.